static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -0,0 +1,198 @@
+import React, {useState} from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import AvatarList from 'sentry/components/avatar/avatarList';
+import TeamAvatar from 'sentry/components/avatar/teamAvatar';
+import Badge from 'sentry/components/badge/badge';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import {CheckWrap} from 'sentry/components/compactSelect/styles';
+import UserBadge from 'sentry/components/idBadge/userBadge';
+import {InnerWrap, LeadingItems} from 'sentry/components/menuListItem';
+import {Tooltip} from 'sentry/components/tooltip';
+import {t} from 'sentry/locale';
+import type {Team} from 'sentry/types/organization';
+import type {User} from 'sentry/types/user';
+import {useTeamsById} from 'sentry/utils/useTeamsById';
+import {useUser} from 'sentry/utils/useUser';
+import type {DashboardDetails, DashboardPermissions} from 'sentry/views/dashboards/types';
+
+interface EditAccessSelectorProps {
+  dashboard: DashboardDetails;
+  onChangeEditAccess?: (newDashboardPermissions?: DashboardPermissions) => void;
+}
+
+/**
+ * Dropdown multiselect button to enable selective Dashboard editing access to
+ * specific users and teams
+ */
+function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorProps) {
+  const currentUser: User = useUser();
+  const dashboardCreator: User | undefined = dashboard.createdBy;
+  const {teams} = useTeamsById();
+  const teamIds: string[] = Object.values(teams).map(team => team.id);
+
+  const [newDashboardPermissions, setNewDashboardPermissions] = useState<
+    DashboardPermissions | undefined
+  >(dashboard.permissions ? structuredClone(dashboard.permissions) : undefined);
+
+  const [selectedOptions, setselectedOptions] = useState<string[]>(
+    dashboard.permissions?.isCreatorOnlyEditable
+      ? ['_creator']
+      : ['_everyone', '_creator', ...teamIds]
+  );
+
+  let isEverythingSelected =
+    selectedOptions.length === ['_everyone', '_creator', ...teamIds].length;
+
+  // Dashboard creator option in the dropdown
+  const makeCreatorOption = () => ({
+    value: '_creator',
+    label: (
+      <UserBadge
+        avatarSize={18}
+        user={dashboardCreator}
+        displayName={
+          <StyledDisplayName>
+            {dashboardCreator?.id === currentUser.id
+              ? `You (${currentUser.email})`
```

Comment:
In this case we should probably use `tct`, it would be applied like so

```suggestion
              ? tct("You ([email])", {email: currentUser.email})
```